lib/controllers/object.js

```diff
@@ -354,6 +355,57 @@ exports.putObjectCopy = async function putObjectCopy(ctx) {
   }
 };
 
+exports.getObjectTagging = async function getObjectTagging(ctx) {
+  const key = ctx.params.key;
+  const exists = await ctx.store.existsObject(ctx.params.bucket, key);
+  if (!exists) {
+    throw new S3Error("NoSuchKey", "The specified key does not exist.", {
+      Key: key
+    });
+  }
+
+  const config = await ctx.store.getSubresource(
+    ctx.params.bucket,
+    key,
+    "tagging"
+  );
+
+  ctx.type = "application/xml";
+  ctx.body = (config || TaggingConfiguration.empty).toXML();
+};
+
+/**
+ * PUT Object tagging
+ * This implementation of the PUT operation uses the tagging subresource to add a set of tags
+ * to an existing object.
+ * {@link https://docs.aws.amazon.com/AmazonS3/latest/API/RESTObjectPUTtagging.html}
+ */
+exports.putObjectTagging = async function putObject(ctx) {
+  const key = ctx.params.key;
+  const bucket = ctx.params.bucket;
+  const exists = await ctx.store.existsObject(bucket, key);
+  if (!exists) {
+    throw new S3Error("NoSuchKey", "The specified key does not exist.", {
+      Key: key
+    });
+  }
+
+  try {
+    await utf8BodyParser(ctx);
+    const config = TaggingConfiguration.validate(ctx.request.body);
+    await ctx.store.putSubresource(bucket, key, config);
+    ctx.body = "";
+  } catch (err) {
+    ctx.logger.error(
+      'Error tagging object "%s" in bucket "%s"',
+      key,
+      bucket,
+      err
+    );
+    throw err;
+  }
+};
+
```

Comment:
Could you insert these definitions in the order as they appear in this list here?
https://docs.aws.amazon.com/AmazonS3/latest/API/RESTObjectOps.html

The structure under `controllers/` is meant to mirror the official REST API docs.